backend/models.py

```diff
@@ -34,10 +34,11 @@ class Folder(models.Model):
     parent = models.ForeignKey('self', on_delete=models.CASCADE, null=True, blank=True)
     path = models.CharField(max_length=200)
     name = models.CharField(max_length=200)
+    is_entry = models.BooleanField(default=False)
 
     class Meta:
         constraints = [
-            models.UniqueConstraint(fields=['path', 'name'], name='folder path constraint')
+            models.UniqueConstraint(fields=['path', 'name', 'is_entry'], name='folder path constraint')
```

Comment:
Change this back. Otherwise you would allow folder to have the same name and path if one is entry and the other isn't.